app/src/lib/git/core.ts

```diff
@@ -182,9 +182,14 @@ function getDescriptionForError(error: DugiteError): string {
     case DugiteError.SSHAuthenticationFailed:
     case DugiteError.SSHPermissionDenied:
     case DugiteError.HTTPSAuthenticationFailed:
-      return `Authentication failed. You may not have permission to access the repository or the repository may have been archived. Open ${
-        __DARWIN__ ? 'preferences' : 'options'
-      } and verify that you're signed in with an account that has permission to access this repository.`
+      return `Authentication failed. Some common reasons include:
+- You are not logged in to your account: see ${
+        __DARWIN__ ? 'Preferences' : 'Options'
+      }
+- You may need to logout and login to refresh your token.
+- You do not have permission to access this repository.
+- The repository is archived on GitHub. Check the repository settings to confirm you are still permitted to push commits.
+- If you use SSH authentication, check your key is added to the SSH agent and associated with your account.`
```

Comment:
```suggestion
- If you use SSH authentication, check that your key is added to the ssh-agent and associated with your account.`
```

Minor nitpick but since we refer to it as `ssh-agent` throughout our help docs I changed this to reflect that.